### PR TITLE
Day 1 - Sonar Sweep

### PR DESCRIPTION
**Pull request by luisandani**

### CODE REVIEW

File: 01/main.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	counter := 0
+	for scan.Scan() {
+		newDepth, err := strconv.Atoi(scan.Text())
+		if err != nil {
+			fmt.Printf("Error converting text to int. %s\n", err)
+		}
+
+		if depth == 0 {
+			fmt.Printf("%d (N/A - no previous measurement)\n", newDepth)
+		} else {
```
Comment (stefanoj3) [inline 01/main.go]: Can you think of a way to write this code without the else statement?

Not an error, but the more indentation the harder to read, as a rule of thumb I try to stay away from else statements as much as possible.

File: 01/main.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	depth := 0
+	counter := 0
+	for scan.Scan() {
+		newDepth, err := strconv.Atoi(scan.Text())
+		if err != nil {
+			fmt.Printf("Error converting text to int. %s\n", err)
+		}
+
+		if depth == 0 {
+			fmt.Printf("%d (N/A - no previous measurement)\n", newDepth)
```
Comment (stefanoj3) [inline 01/main.go]: does this ever happen?

File: 01/main.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			}
+			fmt.Printf("%d %s\n", newDepth, op)
+		}
+
+		depth = newDepth
+	}
+
+	// check for errors
+	if err := scan.Err(); err != nil {
+		fmt.Printf("Scanner contained an error: %s\n", err)
```
Comment (stefanoj3) [inline 01/main.go]: shouldn't the program stop if we could not read everything?

**Review by stefanoj3 — COMMENTED** [review]
When you read my comments keep in mind I didn't run the code.

File: 01/main.go (hunk 3 above)
Comment (luisandani) [reply 01/main.go]: yep, that was the first exercise so I had too much debug added. I'll remove it and make it simpler

File: 01/main.go (hunk 2 above)
Comment (luisandani) [reply 01/main.go]: it happens onthe first iteration, only needed for me at the beginning for debugging purposes.

File: 01/main.go (hunk 1 above)
Comment (luisandani) [reply 01/main.go]: it will get simplified in the next commit

**Pull request merged**